### PR TITLE
Webpack auto-transpilation plugin

### LINKED ISSUE #9: Support magic import path for "generated" schemas
The generated shapeshifter file (usually an index.js file) encounters a ton of churn any time a schema changes. This can cause merge conflicts when 2 separate branches generate a new output file.

Perhaps support a magic import path, like `import  { SomeShape } from 'shapeshifter/schemas';`, which will automatically generate the shapeshifter file during a build process (webpack), instead of including the generated file in the source code.

Pros:
* No build command needs to be run
* No file committed to source
* No possibility of conflicts

Cons:
* Breaks IDE/editor import lookups and references
* Not apparent whether a schema/type/constant is supported (unless looking for the JSON files)

### PR DESCRIPTION
**Pull request by milesj**
cc: @kesne 

A webpack plugin that utilizes a fake import path (`shapeshifter/schematics`) to auto-transpile the schematic files during the webpack bundling process.

Fixes #9

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,8 +56,10 @@
   },
   "devDependencies": {
     "@milesj/build-tool-config": "^0.9.0",
+    "babel-loader": "^7.0.0",
     "prop-types": "^15.5.10",
-    "typescript": "^2.3.4"
+    "typescript": "^2.3.4",
```
Comment (kesne) [inline package.json]: Why is this included?

File: src/bundlers/WebpackResolvePlugin.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    this.importPath = importPath;
+    this.schematicsPath = schematicsPath;
+    this.transpiler = new Transpiler(opts);
+  }
+
+  apply(compiler) {
+    // Start transpiling the schematics and write the output to a temporary file
+    this.transpiler.transpile(this.schematicsPath)
+      .then(source => fs.writeFileSync(TEMP_RESOLVER_FILE, source))
```
Comment (kesne) [inline src/bundlers/WebpackResolvePlugin.js]: Rather than manually writing out the file, could you instead use webpack's FS to do this?

Comment (kesne) [inline src/bundlers/WebpackResolvePlugin.js]: If you use webpackDevServer, for example, it never writes out to the FS, it keeps everything in memory.

**Comment by kesne** [comment]
I'm curious if you should include an empty `schematics.js` file so that eslint won't yell when you import from `shapeshifter/schematics` if you're using the import plugin. Maybe the file could throw an error because that would mean the plugin isn't being used?

File: src/bundlers/WebpackResolvePlugin.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          result.request = TEMP_RESOLVER_FILE;
+        }
+
+        return callback(null, result);
+      });
+    });
+
+    // Delete the temporary file
+    compiler.plugin('compilation', (compilation) => {
+      compilation.plugin('succeed-module', (result) => {
```
Comment (kesne) [inline src/bundlers/WebpackResolvePlugin.js]: Will this cause the file to get unlinked during dev builds?

File: package.json (hunk 1 above)
Comment (milesj) [inline package.json]: Because it's used to validate the generated TS files.

File: src/bundlers/WebpackResolvePlugin.js (hunk 2 above)
Comment (milesj) [inline src/bundlers/WebpackResolvePlugin.js]: Yeah most likely, was using it more as a proof of concept. Still need to do some cleanup stuff.

**Comment by milesj** [comment]
@kesne Updated to use `outputFileSystem`. Should now work with devserver and other webpack features.

**Pull request merged**